fedimint-server/src/lib.rs

```diff
@@ -68,217 +68,193 @@ pub trait HasApiContext<State> {
     ) -> (&State, ApiEndpointContext<'_>);
 }
 
-/// Main server for running Fedimint consensus and APIs
-#[derive(Debug)]
-pub struct FedimintServer {
-    /// Location where configs are stored
-    pub data_dir: PathBuf,
-    /// Module and endpoint settings necessary for starting the API
-    pub settings: ConfigGenSettings,
-    /// Database shared by the API and consensus
-    pub db: Database,
-    /// Version hash
-    pub version_hash: String,
-}
-
-impl FedimintServer {
-    pub fn new(
-        data_dir: PathBuf,
-        settings: ConfigGenSettings,
-        db: Database,
-        version_hash: String,
-    ) -> Self {
-        Self {
-            data_dir,
-            settings,
-            db,
-            version_hash,
+pub async fn run(
```

Comment:
nit: now that it's not a member anymore the fn should have a more descriptive name